plenum/recorder/replayable_node.py

```diff
@@ -50,12 +50,20 @@ def consume_req_queue_for_pre_prepare(self, ledger_id, view_no,
             # never applied as dynamic validation is a read only
             # operation, it is functionally correct.
             # Can be fixed by capturing the exact order
+            idx = 0
+            reqs = []
             for req_id in req_ids:
                 key = req_id
                 fin_req = fin_reqs[key]
                 self.processReqDuringBatch(
-                    fin_req, tm, valid_reqs, invalid_reqs, rejects)
-
+                    fin_req, tm, idx, reqs, rejects)
+                idx += 1
+            invalid_indices = [idx for _, idx, _ in rejects]
+            for ind, req in enumerate(reqs):
+                if ind in invalid_indices:
+                    invalid_reqs.append(req)
+                else:
+                    valid_reqs.append(req)
             return valid_reqs, invalid_reqs, rejects, tm
```

Comment:
The method should return exactly the same output as replica's one, so it needs to be `reqs, rejects, tm`